recipe-generator/src/main/resources/templates/ts/ingredient.liquid

```diff
@@ -19,7 +19,9 @@ class {{ingredientName}} extends {{superclass}} {
         {%- for param in initializer.params -%}
             {%- assign type = info.requiredTypes[param] -%}
             {% raw %} {% endraw %}
-            {%- if type == 'string' or type == 'int' or type == 'float' or type == 'boolean' -%}
+            {%- if type == 'string' -%}
+                && (args[{{forloop.index0}}] === null || typeof args[{{forloop.index0}}] === "string")
```

Comment:
This is just the javascript constructor logic that says if you see a null, then assume it's a string argument and match it against the right signature.